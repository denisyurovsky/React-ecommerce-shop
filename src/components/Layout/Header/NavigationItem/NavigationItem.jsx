import PropTypes from 'prop-types';
import React from 'react';

import styles from './NavigationItem.module.scss';

export const NavigationItem = ({ children }) => {
  return (
    <li className={styles.item}>
      <a href="/">{children}</a>
    </li>
  );
};

NavigationItem.propTypes = {
  children: PropTypes.node,
};
