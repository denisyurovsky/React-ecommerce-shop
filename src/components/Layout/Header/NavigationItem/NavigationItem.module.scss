@import "../../../../assets/styles/colors.scss";

.item {
  display: flex;
  align-items: center;
  padding: 20px;

  a {
    text-decoration: none;
    color: white;
  }
}

.item:hover {
  background-color: rgba(#fff, 0.15);
  cursor: pointer;
}
