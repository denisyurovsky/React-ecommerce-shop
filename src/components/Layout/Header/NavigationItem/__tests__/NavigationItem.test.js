import { render } from '@testing-library/react';
import React from 'react';

import { NavigationItem } from '../NavigationItem';

describe('NavigationItem component', () => {
  describe('snapshots', () => {
    it('renders a valid snapshot', () => {
      const { asFragment } = render(
        <NavigationItem>
          Text in block test
          <div>Div test</div>
          <p>Paragraph test</p>
          <span>Span test</span>
        </NavigationItem>
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
