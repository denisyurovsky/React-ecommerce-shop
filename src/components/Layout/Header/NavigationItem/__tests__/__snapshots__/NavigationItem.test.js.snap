// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem component snapshots renders a valid snapshot 1`] = `
<DocumentFragment>
  <li
    class="item"
  >
    <a
      href="/"
    >
      Text in block test
      <div>
        Div test
      </div>
      <p>
        Paragraph test
      </p>
      <span>
        Span test
      </span>
    </a>
  </li>
</DocumentFragment>
`;
